
<!--==========================================================================
	FTL macros 
	========================================================================== -->


<!--==========================================================================
	CSS styles 
	========================================================================== -->


<style type="text/css">

body {
	background-color: #dddddd;
}

/* Table styles. */

table {
  border-color: #000;
  border-spacing: 0px;
  border-style: solid;
  border-width: 1px;
  cell-spacing: 0px;
}

.noBorder {
  border-width: 0px;
}

td, th {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 10pt;
  padding: 2px 0.5em;
  white-space: nowrap;
}

td.numeric {
  text-align: right;
}

th {
  background-color: #c0c0c0;
}

th.mainHeader {
  background-color: #808080;
  color: #ffffff;
  text-align: left;
}

th a {
  color: #000080;
  text-decoration: none;
}

th a:visited {
  color: #000080;
}

th a:active, th a:hover {
  color: #800000;
  text-decoration: underline;
}

.toc {
	border: 1px solid #aaa;
	background-color: #eeeeee;
	padding: 5px;
	font-size: 95%;
}

.note {
	border: 1px solid #aaa;
	background-color: #eeeeee;
	padding: 5px;
	font-size: 90%;
	text-align: left;
}

.main {
	margin-top: 15px;
	width: 1000px;
	text-align: left;
	background: #ffffff;
	border: 3px solid #777777;
}

.histo {
	table-layout:fixed;
	width:100%;
	border:1px solid #aaa;
	word-wrap:break-word;
}

</style>


<!--==========================================================================
	Index
	========================================================================== -->

<center>
<div class="main">

<center> <h3> SnpEff: Variant analysis </h3> </center>

<div style="margin-left: .5em">
<table class="toc"><tr><td>
	<center><b>Contents</b></center>
	<a href="#summary">Summary</a><br>
	<a href="#changeRateChr"> Variant rate by chromosome</a><br>
	<a href="#changesByType">Variants by type</a><br>
	<a href="#effectsImpact"> Number of variants by impact </a><br>
	<a href="#effectsImpact"> Number of variants by functional class </a><br>
	<a href="#effects"> Number of variants by effect </a><br>
	<a href="#quality">Quality histogram</a><br>
	<a href="#indels">InDel length histogram</a><br>
	<a href="#baseChages">Base variant table</a><br>
	<a href="#tstv">Transition vs transversions (ts/tv)</a><br>
	<a href="#alleleFreq"> Allele frequency </a><br>
	<a href="#alleleCount"> Allele Count </a><br>
	<a href="#codonChanges"> Codon change table </a><br>
	<a href="#aaChanges"> Amino acid change table </a><br>
	<a href="#chrChanges"> Chromosome variants plots </a><br>
	<a href="snpEff_genes.txt"> Details by gene </a><br>
</tr></td></table>
</div>


<!--==========================================================================
	Summary table 
	========================================================================== -->

<hr>
<a name="summary">
<center>
<b>Summary</b><p>

<table border=0>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> Genome </b> </td>
		<td> GRCh37.75 </td>
	</tr>
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Date </b> </td>
		<td> 2022-12-15 13:35 </td>
	</tr>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> SnpEff version </b> </td>
		<td> <pre>SnpEff 5.1d (build 2022-04-19 15:49), by Pablo Cingolani</pre> </td>
	</tr>
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Command line arguments </b> </td>
		<td> <pre>SnpEff  GRCh37.75 ../data/NA12878.chrom11.ILLUMINA.bwa.CEU.low_coverage.20121211/analysis/all.vcf 
</pre> </td>
	</tr>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> Warnings </b> </td>
		   
		<td bgcolor="#ffff00"> 46 </td>
	</tr>	
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Errors </b> </td>
		 
		<td bgcolor="#dddddd"> 0 </td>
	</tr>	
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of lines (input file) </b> </td>
		<td> 114 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of variants (before filter) </b> </td>
		<td> 112 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of not variants <br>(i.e. reference equals alternative) </b> </td>
		<td> 0 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of variants processed <br> (i.e. after filter and non-variants) </b> </td>
		<td> 112 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of known variants <br>(i.e. non-empty ID) </b> </td>
		<td> 
		105
		( 93.75% ) 
		</td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of multi-allelic VCF entries <br>(i.e. more than two alleles) </b> </td>
		<td> 
		0
		</td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of effects </b> </td>
		<td> 825 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Genome total length </b> </td>
		<td> 32,036,512,384 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Genome effective length </b> </td>
		<td> 135,006,516 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Variant rate </b> </td>
		<td> 1 variant every 1,205,415 bases </td>
	</tr>
</table>
<p>
</center>

<!--==========================================================================
	Variant rate by chromosome
	========================================================================== -->

<hr> 
<a name="changeRateChr"> 

<center>
<b> Variants rate details </b><p>

<table border=1>
	<tr><th> Chromosome </th><th> Length </th><th> Variants </th><th> Variants rate </th></tr> 
		<tr>
			<td> 11 </td>
			<td class="numeric"> 135,006,516 </td>
			<td class="numeric"> 112 </td>
			<td class="numeric"> 1,205,415 </td>
		</tr> 
	<tr>
		<th> Total </th>
		<th class="numeric"> 135,006,516 </th>
		<th class="numeric"> 112 </th>
		<th class="numeric"> 1,205,415 </th>
	</tr> 
</table> 
</center>

<!--==========================================================================
	Variants by type
	========================================================================== -->

<hr> 
<a name="changesByType"> 
<center>
<b> Number variants by type</b><p>

<table border=1>
	<thead>
		<tr>
			<th> <b> Type   </b> </th>
			<th> <b> Total  </b> </th>
		</tr>
	</thead>
    <tbody>
	    <tr>
	    	<td> <b> SNP </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> MNP </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> INS </b> </td>
	    	<td class="numeric" bgcolor="#20de00"> 12 </td>
	    </tr>
	    <tr>
	    	<td> <b> DEL </b> </td>
	    	<td class="numeric" bgcolor="#ff0000"> 88 </td>
	    </tr>
	    <tr>
	    	<td> <b> MIXED </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> INV </b> </td>
	    	<td class="numeric" bgcolor="#00ff00"> 1 </td>
	    </tr>
	    <tr>
	    	<td> <b> DUP </b> </td>
	    	<td class="numeric" bgcolor="#08f600"> 4 </td>
	    </tr>
	    <tr>
	    	<td> <b> BND </b> </td>
	    	<td class="numeric" bgcolor="#11ed00"> 7 </td>
	    </tr>
	    <tr>
	    	<td> <b> INTERVAL </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
    </tbody>
    <tfoot>
	    <tr>
	    	<th><b>Total </b> </th>
	    	<th class="numeric"> 112 </th>
	    </tr>
    </tfoot>
</table>
</center>

<!--==========================================================================
	Effects by impact
	========================================================================== -->

<hr> 
<a name="effectsImpact"> 
<center>
<b> Number of effects by impact </b> <p> 

<table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> HIGH </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 422 </td> 
			<td class="numeric" bgcolor="#ff0000"> 51.152% </td>
		</tr>
		<tr> 
			<td> <b> LOW </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 57 </td> 
			<td class="numeric" bgcolor="#00ff00"> 6.909% </td>
		</tr>
		<tr> 
			<td> <b> MODIFIER </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#c93500"> 346 </td> 
			<td class="numeric" bgcolor="#c93500"> 41.939% </td>
		</tr>
</table><br>
<p>
</center>

<!--==========================================================================
	Effects by functional class
	========================================================================== -->

<hr> 
<a name="effectsFuncClass"> 
<center>
<b> Number of effects by functional class </b> <p> 

<table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
</table><br>
<p>

Missense / Silent ratio: 0

</center>

<!--==========================================================================
	Effects stats
	========================================================================== -->

<hr> 
<a name="effects"> 
<center>
<b> Number of effects by type and region </b> <p> 

<table border=0>
	<tr>
		<th> Type </th>
		<th> Region </th>
	</tr>
	<tr>
		<td> <table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> 3_prime_UTR_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.106% </td>
		</tr>
		<tr> 
			<td> <b> bidirectional_gene_fusion </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 264 </td> 
			<td class="numeric" bgcolor="#ff0000"> 27.966% </td>
		</tr>
		<tr> 
			<td> <b> downstream_gene_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#30ce00"> 51 </td> 
			<td class="numeric" bgcolor="#30ce00"> 5.403% </td>
		</tr>
		<tr> 
			<td> <b> exon_loss_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.106% </td>
		</tr>
		<tr> 
			<td> <b> feature_fusion </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#35c900"> 56 </td> 
			<td class="numeric" bgcolor="#35c900"> 5.932% </td>
		</tr>
		<tr> 
			<td> <b> frameshift_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#639b00"> 104 </td> 
			<td class="numeric" bgcolor="#639b00"> 11.017% </td>
		</tr>
		<tr> 
			<td> <b> gene_fusion </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#926c00"> 152 </td> 
			<td class="numeric" bgcolor="#926c00"> 16.102% </td>
		</tr>
		<tr> 
			<td> <b> intergenic_region </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#32cc00"> 53 </td> 
			<td class="numeric" bgcolor="#32cc00"> 5.614% </td>
		</tr>
		<tr> 
			<td> <b> intron_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#bb4300"> 194 </td> 
			<td class="numeric" bgcolor="#bb4300"> 20.551% </td>
		</tr>
		<tr> 
			<td> <b> non_coding_transcript_exon_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#01fd00"> 3 </td> 
			<td class="numeric" bgcolor="#01fd00"> 0.318% </td>
		</tr>
		<tr> 
			<td> <b> splice_acceptor_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#01fd00"> 3 </td> 
			<td class="numeric" bgcolor="#01fd00"> 0.318% </td>
		</tr>
		<tr> 
			<td> <b> splice_donor_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#01fd00"> 3 </td> 
			<td class="numeric" bgcolor="#01fd00"> 0.318% </td>
		</tr>
		<tr> 
			<td> <b> splice_region_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#03fb00"> 5 </td> 
			<td class="numeric" bgcolor="#03fb00"> 0.53% </td>
		</tr>
		<tr> 
			<td> <b> transcript_ablation </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 2 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.212% </td>
		</tr>
		<tr> 
			<td> <b> upstream_gene_variant </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#31cd00"> 52 </td> 
			<td class="numeric" bgcolor="#31cd00"> 5.508% </td>
		</tr>
</table><br>
 </td>
		<td> <table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> DOWNSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#1ee000"> 51 </td> 
			<td class="numeric" bgcolor="#1ee000"> 6.182% </td>
		</tr>
		<tr> 
			<td> <b> EXON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.121% </td>
		</tr>
		<tr> 
			<td> <b> GENE </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 416 </td> 
			<td class="numeric" bgcolor="#ff0000"> 50.424% </td>
		</tr>
		<tr> 
			<td> <b> INTERGENIC </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#42bc00"> 109 </td> 
			<td class="numeric" bgcolor="#42bc00"> 13.212% </td>
		</tr>
		<tr> 
			<td> <b> INTRON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#748a00"> 190 </td> 
			<td class="numeric" bgcolor="#748a00"> 23.03% </td>
		</tr>
		<tr> 
			<td> <b> SPLICE_SITE_ACCEPTOR </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 2 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.242% </td>
		</tr>
		<tr> 
			<td> <b> SPLICE_SITE_DONOR </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.121% </td>
		</tr>
		<tr> 
			<td> <b> SPLICE_SITE_REGION </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.121% </td>
		</tr>
		<tr> 
			<td> <b> TRANSCRIPT </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 2 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.242% </td>
		</tr>
		<tr> 
			<td> <b> UPSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#1fdf00"> 52 </td> 
			<td class="numeric" bgcolor="#1fdf00"> 6.303% </td>
		</tr>
</table><br>
 </td>
	</tr>
</table>
<p>

<img src="http://chart.apis.google.com/chart?cht=bvs&chs=800x300&chbh=50,4,8&chxt=y,y,x,x&chtt=Variations&chxp=1,50.0|3,50.0&chxr=0,0.0,23.03030303030303|1,0.0,100.0|3,0.0,100.0&chco=00FFFF,FA8072,FF0000,A52A2A,FFA500,DEB887,E9967A,7FFFD4,808080&chdl=Upstream|5%27UTR|Exon|Splice+Donor|Intron|Splice+Acceptor|3%27UTR|Downstream|Intergenic&chxl=1:|%25|2:|Intergenic|Up|5%27UTR|Exon|Donor|Intron|Acceptor|Exon|3%27UTR|Down|Intergenic|3:|&chd=e:AARhAAAAAAAAAAAAAAAAAA,AAAAAAAAAAAAAAAAAAAAAA,AAAAAAAWAAAAAAAWAAAAAA,AAAAAAAAAWAAAAAAAAAAAA,AAAAAAAAAA..AAAAAAAAAA,AAAAAAAAAAAAArAAAAAAAA,AAAAAAAAAAAAAAAAAAAAAA,AAAAAAAAAAAAAAAAAARLAA,ktAAAAAAAAAAAAAAAAAAkt" border=1><p>

</center>

<!--==========================================================================
	Quality
	========================================================================== -->

<hr> 
<a name="quality"> 
<b> Quality:</b> 
<p>
<pre>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>0</td> </tr>
		<tr> <th>Max</th><td>625</td> </tr>
		<tr> <th>Mean</th><td>253.308</td> </tr>
		<tr> <th>Median</th><td>230</td> </tr>
		<tr> <th>Standard deviation</th><td>191.376</td> </tr>
		<tr> <th>Values</th><td>0,36,92,96,113,117,118,173,213,222,223,237,246,247,277,297,378,397,488,560,625</td> </tr>
		<tr> <th>Count</th><td>3,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,2,2</td> </tr>
		</tr>
	</table>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Quality+histogram&chxr=0,1.0,3.0|1,0.0,100.0|3,0.0,100.0&chd=e:..AAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAgAgA&chxl=1:|Count|2:|0||||||||||223||||||||||625|3:|Quality"><br>
</pre>

<!--==========================================================================
	InDels
	========================================================================== -->

<hr> 
<a name="indels"> 
<b> Insertions and deletions length:</b> 
<p>
<pre>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>1</td> </tr>
		<tr> <th>Max</th><td>89</td> </tr>
		<tr> <th>Mean</th><td>5.96</td> </tr>
		<tr> <th>Median</th><td>1</td> </tr>
		<tr> <th>Standard deviation</th><td>15.739</td> </tr>
		<tr> <th>Values</th><td>1,19,20,23,27,36,38,39,51,65,81,89</td> </tr>
		<tr> <th>Count</th><td>88,1,2,1,1,1,1,1,1,1,1,1</td> </tr>
		</tr>
	</table>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chg=8.0,8.0,3,2&chxp=1,50.0|3,50.0&chd=e:..AAApAAAAAAAAAAAAAAAAAA&chxr=0,1.0,88.0|1,0.0,100.0|3,0.0,100.0&chtt=Insertion+deletion+length+histogram&chxl=1:|Count|2:|1||||||||||81|89|3:|Length"><br>
</pre>

<!--==========================================================================
	Base changes
	========================================================================== -->

<hr> 
<a name="baseChages"> 
<center>
<b> Base changes (SNPs) </b> <p>

<table border=1>
    <tr>
    	<td> &nbsp; </td>  <th> <b> A </b> </th> <th> <b> C </b> </th> <th> <b> G </b> </th> <th> <b> T </b> </th>
    </tr>
	    <tr> <th> <b> A </b> </th><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td> </tr>
	    <tr> <th> <b> C </b> </th><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td> </tr>
	    <tr> <th> <b> G </b> </th><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td> </tr>
	    <tr> <th> <b> T </b> </th><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="ffffff"> 0 </td> </tr>
</table>
</center>

<p>

<!--==========================================================================
	Ts/Tv
	========================================================================== -->

<hr> 
<a name="tstv">
<center> <b> Ts/Tv (transitions / transversions) </b> <p> </center>

<small>
<b>Note:</b> Only SNPs are used for this statistic.<br>
<b>Note:</b> This Ts/Tv ratio is a 'raw' ratio (ratio of observed events).
</small>
<p>

<center>
<table border=1>
	<tr> <th> Transitions </th><td class="numeric"> 0 </td> </tr>
	<tr> <th> Transversions </th><td class="numeric"> 0 </td> </tr>
	<tr> <th> Ts/Tv ratio </th><td class="numeric"> 0 </td> </tr>
</table>
</center>
<p>

	<b>All variants:</b>
	<pre>No results available (empty input?)</pre>
	<p>
	<b>Only known variants</b> (i.e. the ones having a non-empty ID field):
	<pre>No results available (empty input?)</pre>
<p>

<!--==========================================================================
	Allele frequency
	========================================================================== -->

<hr> 
<a name="alleleFreq">
<center> <b> Allele frequency </b> <p> </center>

	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chd=e:KP..AA5m&chg=10.0,5.0,3,2&chxp=1,50.0|3,50.0&chtt=Allele+Frequency+%25&chxr=0,1.0,54.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Count|2:|0|||100|3:|Alleles+frequency"><br>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>0</td> </tr>
		<tr> <th>Max</th><td>100</td> </tr>
		<tr> <th>Mean</th><td>67.325</td> </tr>
		<tr> <th>Median</th><td>50</td> </tr>
		<tr> <th>Standard deviation</th><td>31.75</td> </tr>
		<tr> <th>Values</th><td>0,50,75,100</td> </tr>
		<tr> <th>Count</th><td>10,54,1,49</td> </tr>
		</tr>
	</table>
<p>
	
<!--==========================================================================
	Allele Count
	========================================================================== -->

<hr> 
<a name="alleleCount">
<center> <b> Allele Count </b> <p> </center>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chd=e:Lh..8yAAHr&chtt=Allele+Count&chxp=1,50.0|3,50.0&chxl=1:|Count|2:|0||||4|3:|Allele+Count&chxr=0,1.0,49.0|1,0.0,100.0|3,0.0,100.0"><br>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>0</td> </tr>
		<tr> <th>Max</th><td>4</td> </tr>
		<tr> <th>Mean</th><td>1.526</td> </tr>
		<tr> <th>Median</th><td>1</td> </tr>
		<tr> <th>Standard deviation</th><td>0.904</td> </tr>
		<tr> <th>Values</th><td>0,1,2,3,4</td> </tr>
		<tr> <th>Count</th><td>10,49,47,1,7</td> </tr>
		</tr>
	</table>
<p>


<!--==========================================================================
	Hom/Het table
	========================================================================== -->

<hr> 
<a name="homHet">
<center> <b> Hom/Het per sample </b> <p> </center>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chd=e:HC..AAAA&chxp=1,50.0|3,50.0&chtt=Heterozygous+genotypes&chxr=0,0.0,54.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Heterozygous+count|2:|0|||3|3:|Sample+number"><br>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chd=e:WZ..AAAA&chxp=1,50.0|3,50.0&chtt=Homozygous+%28ALT%29+genotypes&chxr=0,0.0,42.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Homozygous+count|2:|0|||3|3:|Sample+number"><br>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chd=e:x6AA....&chxp=1,50.0|3,50.0&chtt=Missing+genotypes&chxr=0,18.0,114.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Missing+count|2:|0|||3|3:|Sample+number"><br>
	
	<pre>Sample_names , NA12878, NA12878_1, NA12878_2, NA12878.chrom11.ILLUMINA.bwa.CEU.low_coverage.20121211.bam
Reference , 0, 0, 0, 0
Het , 6, 54, 0, 0
Hom , 15, 42, 0, 0
Missing , 93, 18, 114, 114
</pre>
<p>

<!--==========================================================================
	Codon change table
	========================================================================== -->
	
<hr> 
<a name="codonChanges">
<center>
<b> Codon changes</b> <p>

	<div class="note">
		How to read this table: <br>
		- Rows are reference codons and columns are changed codons. E.g. Row 'AAA' column 'TAA' indicates how many 'AAA' codons have been replaced by 'TAA' codons.<br>
		- Red background colors indicate that more changes happened (heat-map).<br>
		- Diagonals are indicated using grey background color <br> 
		- WARNING: This table may include different translation codon tables (e.g. mamalian DNA and mitochondrial DNA).<br>
		<p>
	</div><p>

  
<table border=1>
	<thead>
		<tr> 
			<th> &nbsp; </th>
			
		</tr>
	</thead>
</table>
</center>

<!--==========================================================================
	Amino acid change table
	========================================================================== -->

<hr> 
<a name="aaChanges">
<center>
<b> Amino acid changes</b><p> 

	<div class="note">
		How to read this table: <br>
		- Rows are reference amino acids and columns are changed amino acids. E.g. Row 'A' column 'E' indicates how many 'A' amino acids have been replaced by 'E' amino acids.<br>
		- Red background colors indicate that more changes happened (heat-map).<br>
		- Diagonals are indicated using grey background color <br> 
		- WARNING: This table may include different translation codon tables (e.g. mamalian DNA and mitochondrial DNA).<br>
		<p>
	</div><p>

<table border=1>
	<thead>
		<tr> 
			<th> &nbsp; </th>
			
		</tr>
	</thead>
</table>
</center>

<!--==========================================================================
	Chromosome variants table
	========================================================================== -->

<hr> 
<a name="chrChanges">
<center>
<b> Variants by chromosome</b><p> 

<center>
	<pre>
		<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Variants+histogram%3A+11&chxr=0,0.0,8.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Variants%2F1000Kb|2:|0|||||||||||||13000000|||||||||||||26000000|||||||||||||39000000|||||||||||||52000000|||||||||||||65000000|||||||||||||78000000|||||||||||||91000000|||||||||||||104000000|||||||||||||117000000|||||||||||||130000000|||||135000000|3:|Position&chd=e:3rHrHrAAHrHrHrgAAAHrAAHrHrAAHrAAQAHrAAAAAAAAHrQAAAAAAAQAAAXrAAQAAAQAHrAAAAAAAAHrAAQAAAXrAAHrAAHr..QAXrQAAAAAQAHrAAHrQAAAAAXrAAHrQAAAHrAAHrHrAAHrAAAAAAAAAAAAAAAAAAAAHrAAHrAAAAAAAAHrHrAAQAgAQAHrHrAAAAgAAAQAQAQAXrAAAAHrAAHrHrAAAAAAHrAAAAAAAAAAAAAAHrAAHrHrQAAAAAAAAAAAAAAAHrAA"><br>
		11, Position,0,1000000,2000000,3000000,4000000,5000000,6000000,7000000,8000000,9000000,10000000,11000000,12000000,13000000,14000000,15000000,16000000,17000000,18000000,19000000,20000000,21000000,22000000,23000000,24000000,25000000,26000000,27000000,28000000,29000000,30000000,31000000,32000000,33000000,34000000,35000000,36000000,37000000,38000000,39000000,40000000,41000000,42000000,43000000,44000000,45000000,46000000,47000000,48000000,49000000,50000000,51000000,52000000,53000000,54000000,55000000,56000000,57000000,58000000,59000000,60000000,61000000,62000000,63000000,64000000,65000000,66000000,67000000,68000000,69000000,70000000,71000000,72000000,73000000,74000000,75000000,76000000,77000000,78000000,79000000,80000000,81000000,82000000,83000000,84000000,85000000,86000000,87000000,88000000,89000000,90000000,91000000,92000000,93000000,94000000,95000000,96000000,97000000,98000000,99000000,100000000,101000000,102000000,103000000,104000000,105000000,106000000,107000000,108000000,109000000,110000000,111000000,112000000,113000000,114000000,115000000,116000000,117000000,118000000,119000000,120000000,121000000,122000000,123000000,124000000,125000000,126000000,127000000,128000000,129000000,130000000,131000000,132000000,133000000,134000000,135000000
11,Count,7,1,1,0,1,1,1,4,0,1,0,1,1,0,1,0,2,1,0,0,0,0,1,2,0,0,0,2,0,3,0,2,0,2,1,0,0,0,0,1,0,2,0,3,0,1,0,1,8,2,3,2,0,0,2,1,0,1,2,0,0,3,0,1,2,0,1,0,1,1,0,1,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,1,1,0,2,4,2,1,1,0,0,4,0,2,2,2,3,0,0,1,0,1,1,0,0,0,1,0,0,0,0,0,0,0,1,0,1,1,2,0,0,0,0,0,0,0,1,0

	</pre>
</center>

<!--==========================================================================
	Variants by gene 
	========================================================================== -->

<center>
	<hr>
	<b> Details by gene </b><p>
	<p>
	<b><a href="snpEff_genes.txt">Here</a></b> you can find a tab-separated table.
</center>
</div>
</center>